### PR TITLE
KFSPTS-14678 add template batch job for the RASS integration

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<beans xmlns="http://www.springframework.org/schema/beans"
```
Comment (jhulslander) [inline src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml]: the format on this file seems off

File: src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public class RassServiceImpl implements RassService {
+
+	@Override
+	public void readXML() {
+
+	}
+	
+	@Override
+	public boolean updateKFS() {
+		updateAgencies();
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java]: Should we set a boolean like:
boolean successfullyProcessed = updateAgencies();
successfullyProcessed &= updateProposals();
successfullyProcessed &= updateAwards();
return successfullyProcessed;

File: src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package edu.cornell.kfs.rass.batch.service.impl;
+
+import edu.cornell.kfs.rass.batch.service.RassService;
+
+public class RassServiceImpl implements RassService {
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java]: should we have a member variable of this service hold the path to the staging directory, and then set that value in Spring?

File: src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+                           http://www.springframework.org/schema/beans/spring-beans-2.0.xsd
+                           http://www.springframework.org/schema/tx
+                           http://www.springframework.org/schema/tx/spring-tx-2.0.xsd
+                           http://www.springframework.org/schema/aop
+                           http://www.springframework.org/schema/aop/spring-aop-2.0.xsd">
+
+	<bean id="cu-rass" parent="rassModuleConfiguration-parentBean"/>
+		
+	<bean id="rassModuleConfiguration-parentBean" class="org.kuali.kfs.sys.FinancialSystemModuleConfiguration" abstract="true">
```
Comment (jhulslander) [inline src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml]: Should we had a batchFileDirectories list and setup the the staging directory for RASS?

File: src/main/java/edu/cornell/kfs/rass/batch/service/RassService.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package edu.cornell.kfs.rass.batch.service;
+
+public interface RassService {
+	
+	void readXML();
+	
+	boolean updateKFS();
+	
+	boolean updateAgencies();
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/rass/batch/service/RassService.java]: should updateAgencies, updateProposals and updateAwards just be protected functions on the implementations and not be part of the interface?  The job should just call updateKFS right?

File: src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml (hunk 4 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml]: There is a batchFileDirectories defined

File: src/main/java/edu/cornell/kfs/rass/batch/service/RassService.java (hunk 5 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/service/RassService.java]: good point, will update

File: src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java (hunk 3 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java]: will add

File: src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java (hunk 2 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java]: I will add the boolean.

File: src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml (hunk 1 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml]: will format

File: src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml (hunk 4 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/rass/cu-spring-rass.xml]: adding the staging directory

**Comment by db483** [comment]
Hi Jay,

I've added the suggested changes and this is ready for review again. Thank you.

**Pull request merged**